enterprise/reporting/reporting.go

```diff
@@ -90,11 +96,23 @@ func NewDefaultReporter(ctx context.Context, conf *config.Config, log logger.Log
 	maxOpenConnections := config.GetIntVar(32, 1, "Reporting.maxOpenConnections")
 	dbQueryTimeout = config.GetReloadableDurationVar(60, time.Second, "Reporting.dbQueryTimeout")
 	maxReportsCountInARequest := conf.GetReloadableIntVar(10, 1, "Reporting.maxReportsCountInARequest")
+	eventSamplingEnabled := conf.GetReloadableBoolVar(false, "Reporting.eventSampling.enabled")
+	eventSamplingDuration := conf.GetReloadableDurationVar(60, time.Minute, "Reporting.eventSampling.durationInMinutes")
+	eventSamplerType := conf.GetReloadableStringVar("badger", "Reporting.eventSampling.type")
+	eventSamplingCardinality := conf.GetReloadableIntVar(1000000, 1, "Reporting.eventSampling.cardinality")
```

Comment:
@satishrudderstack should we keep default for `Reporting.eventSampling.type` as `badger` or `in_memory_cache`?

And what should we keep default value for `Reporting.eventSampling.cardinality`?

Reply:
Like we discussed offline, let's have `badger` as default. We can switch to in-memory implementation if we see this is adding latency into the sytem.

For max cardinality
Assuming a limit of 1-10MB on the size of map and each entry size of ~50 Bytes or so

5 MB * 1024 KB * 1024 B / 50 B = ~100 K

Reply:
@cisse21 Do we have metrics on how much latency badger adds in processing from deduplication use case ? For our case, I think the badger calls might be of the order of 1/5th to 1/10th of dedup calls. Do you see potential latency issues introduced in the processing due to this ?